common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java

```diff
@@ -125,10 +125,15 @@ public static void load(String originalName, ClassLoader loader) {
             }
         }
 
-        if (url == null) {
-            // Fall back to normal loading of JNI stuff
+        try {
+            // first try to load from java.library.path
             loadLibrary(loader, name, false);
+            logger.info("{} was loaded from java.libary.path", name);
             return;
+        } catch (Throwable ex) {
+            logger.debug(
+                    "{} cannot be loaded from java.libary.path, "
+                    + "now trying export to -Dio.netty.native.workdir: {}", name, WORKDIR, ex);
```

Comment:
the throwable is treated specially, see io.netty.util.internal.logging.MessageFormatter arrayFormat